tests/core/test_state_store.py

```diff
@@ -54,6 +55,7 @@ def test_is_checkpoint_updated():
     assert s2.is_checkpoint_updated() is False
 
     s2.update_state("test-state1")
+    time.sleep(1)
```

Comment:
why is this needed?

Reply:
because `update_state` is an async call. 
Even though I am updating `/subdir1/checkpoint_test1.txt` in line 63 after this call, the actual `update_state` happens after `checkpoint_test1.txt` is written. So while we expect line 66 to be `True`, it will return `False` without the 1 second delay.

I tested this out as follows -
```
tests/core/test_state_store.py::test_is_checkpoint_updated 
last_timestamp: 1586394007.8299778 and timestamp of chkpnt file: 1586394007.8291862
```
`last_timestamp` is when `update_state` writes to file to disk(line 57), and `timestamp of chkpnt file` is when the checkpoint file is written to disk(line64).

Reply:
this is the failed build link - https://tiny.amazon.com/114zqzab7/IsenLink

Reply:
why would it fail? you say update_state() , a new json file would be written. 
Since there is no new checkpoint file created, irrespective of whether update_state wrote the file or not youshould get False as result of is_checkpoint_updated.
Am I missing something?

Reply:
yes you should get False in line 60 and True in line 66. the `time.sleep` is to ensure a True value in line 66.